tests/verilog/past_signedness.ys

```diff
@@ -0,0 +1,17 @@
+logger -expect-no-warnings
+
+read_verilog -formal <<EOT
+module top(input clk);
+    reg signed [3:0] value = -1;
+    reg ready = 0;
+
+    always @(posedge clk) begin
+        if (ready)
+            assert ($past(value) == -1);
```

Comment:
It would be nice if we also ensured that $past of an unsigned expression remains unsigned.

Reply:
Good point, added a second test for that.